src/Chronos.hs

```diff
@@ -1836,6 +1847,40 @@ monthToZeroPaddedDigitBS (Month x) =
 zeroPadDayOfMonthBS :: DayOfMonth -> BB.Builder
 zeroPadDayOfMonthBS (DayOfMonth d) = indexTwoDigitByteStringBuilder d
 
+
+within :: Time -> TimeInterval -> Bool
+t `within` (TimeInterval t0 t1) = t >= t0 && t <= t1 || t >= t1 && t <= t0
```

Comment:
this should assume correct construction of `TimeInterval`, to avoid branching.

E.g.

```
t `within` (TimeInterval t0 t1) = t >= t0 && t <= t1
```